main/src/mill/main/MainRunner.scala

```diff
@@ -220,6 +220,7 @@ class MainRunner(
            |  // doesn't get picked up during reflective child-module discovery
            |  def millSelf = Some(this)
            |
+           |  @scala.annotation.nowarn("cat=deprecation")
```

Comment:
```suggestion
           |  @_root_.scala.annotation.nowarn("cat=deprecation")
```

The `HygeneTests` on Java 17 test fail. I think, we need to use `_root_` here, too.